indico/htdocs/js/indico/modules/contributions/common.js

```diff
@@ -105,48 +105,56 @@
         });
     }
 
+    function changeContribDisplayedStartDate(itemPicker) {
+        itemPicker.closest('tr').find('td.start-date').html($('<em>', {'text': $T.gettext('Not scheduled')}));
+    }
+
     function setupSessionPicker(createURL) {
         $('#contribution-list .session-item-picker').itempicker({
             filterPlaceholder: $T.gettext('Filter sessions'),
             containerClasses: 'session-item-container',
             footerElements: [{
-                title: $T.gettext('Add new session'),
-                onClick: function() {
+                title: $T.gettext('Assign new session'),
+                onClick: function(itemPicker) {
                     ajaxDialog({
                         title: $T.gettext('Add new session'),
                         url: createURL,
                         onClose: function(data) {
                             if (data) {
+                                var postData = {session_id: data.new_session_id};
                                 $('.session-item-picker').itempicker('refreshItemList', data.sessions);
+
+                                patchObject(itemPicker.data('href'), itemPicker.data('method'), postData).then(function(contribData) {
```

Comment:
the call to `patchObject` was redundant since I am already triggering `onSelect` when calling the `selectItem` method